### PR TITLE
Handle blocktests

### PR DESCRIPTION
**Pull request by palashmarantas**
# Issue Link

Added a generic blocktests support for blocklist and quarantined tests

# Description

Added a generic blocktests support for blocklist and quarantined tests

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test With runners and postman
- [ ] Test B

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pkg/core/models.go [hunk 1]
```diff
@@ -340,3 +340,11 @@ const (
...
 	ExecutionTask TaskType = "execute"
 )
+
+// BlockTestType stores block tests type
+type BlockTestType string
+
+const (
+	Blocklisted BlockTestType = "blocklisted"
+	Quarantined BlockTestType = "quarantined"
+)
```
Comment (saurabh-prakash) [inline pkg/core/models.go:350]: why is this required?

**Review by saurabh-prakash — COMMENTED** [review]
@palashmarantas Can you do a `git mv` for `setup.go` file? It is showing as added and removed right now losing the git history.

File: pkg/blocktestservice/setup.go [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+	"github.com/LambdaTest/synapse/pkg/core"
+	"github.com/LambdaTest/synapse/pkg/global"
+	"github.com/LambdaTest/synapse/pkg/lumber"
+)
+
+const (
+	delimiter = "##"
+)
+
+//blocktest represents the blocklisted test suites and test cases.
```
Comment (saurabh-prakash) [inline pkg/blocktestservice/setup.go:25]:
```suggestion
// blocktest represents the blocklisted test suites and test cases.
```

File: pkg/blocktestservice/setup.go [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+			Timeout: 15 * time.Second,
+			Transport: &http.Transport{
+				DisableKeepAlives: true,
+			},
+		}}, nil
+}
+
+//fetchBlockListFromNeuron
+func (tbs *TestBlockListService) fetchBlockListFromNeuron(ctx context.Context, repoID, branch string) error {
+
```
Comment (rishabh-arya95) [inline pkg/blocktestservice/setup.go:75]: nit: remove empty new line after function starts, you can use go-lint

Comment (rishabh-arya95) [inline pkg/blocktestservice/setup.go:75]: nit: remove empty new line after function starts, you can use go-lint

File: pkg/blocktestservice/setup.go [hunk 4]
```diff
@@ -0,0 +1,197 @@
+// Package blocktestservice is used for creating the blocklist file
+package blocktestservice
```
Comment (rishabh-arya95) [inline pkg/blocktestservice/setup.go:2]: use git mv command to do renaming

Comment (rishabh-arya95) [inline pkg/blocktestservice/setup.go:2]: use git mv command to do renaming

Comment (palashmarantas) [reply pkg/blocktestservice/setup.go:2]: tried restoring using git mv but did not work out

File: pkg/blocktestservice/setup.go [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+func (tbs *TestBlockTestService) populateBlockList(blocktestSource string, blocktestLocators []*blocktestLocator) {
+
+	i := 0
+	for _, test := range blocktestLocators {
+		//locators must end with delimiter
+		if !strings.HasSuffix(test.Locator, delimiter) {
+			test.Locator += delimiter
+		}
+		i = strings.Index(test.Locator, delimiter)
+		//TODO: handle duplicate entries and ignore its individual suites or testcases in blocklist if file is blocklisted
```
Comment (VikrantKS) [inline pkg/blocktestservice/setup.go:186]: Following code will lead to duplicate locators and we are not overriding source , we need to give precedence to api over yml

**Review by VikrantKS — CHANGES_REQUESTED** [review]

**Review by rishabh-arya95 — APPROVED** [review]

File: pkg/core/models.go (hunk 1 above)
Comment (palashmarantas) [reply pkg/core/models.go]: outdated

**Comment by saurabh-prakash** [comment]
Trusting the 2 reviewers. Merging.
